### PR TITLE
Implemented propagate backward

### LINKED ISSUE #2998: Propagate to frame before
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour
Propagation is allowed to frames near the current frame, both backwards and forwards

### Current Behaviour
Only forward propagation is allowed

### Steps to Reproduce (for bugs)
1. Goto frame 10 of any job
2. Create a shape/track
3. Save the job
4. Select the annotation, and try to propagate (Ctrl+B)
5. Only positive integers are accepted (forward propagation)

### Context
Sometime, the first few frames are blurry, and it's easier to start from a non-1st frame, and then propagate the shape back to the frames that are blurry.

### Your Environment

- Server version: 1.2
- Core version: 3.10.0
- Canvas version: 2.2.1
- UI version: 1.13.3

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #2998

image

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Review by klakhov — COMMENTED** [review]
I think this slider blinking should be fixed:
propagate1

Also what do you think about showing current `from` value in input(maybe we should disable it in such case) instead of keeping it empty?
image

File: cvat-ui/src/actions/annotation-actions.ts [hunk 1]
```diff
@@ -423,9 +445,10 @@ export function propagateObjectAsync(sessionInstance: any, objectState: any, fro
             });
 
             const copy = getCopyFromState(objectState);
-            await sessionInstance.logger.log(LogType.propagateObject, { count: to - from + 1 });
+            await sessionInstance.logger.log(LogType.propagateObject, { count: Math.abs(to - from) });
             const states = [];
-            for (let frame = from; frame <= to; frame++) {
+            const sign = Math.sign(to - from);
+            for (let frame = from + sign; sign > 0 ? frame <= to : frame >= to; frame += 1 * sign) {
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-ui/src/actions/annotation-actions.ts:451]: There is no logic in ``1*`` behined.
BTW, Math.sign also can return 0.

**Comment by bsekachev** [comment]
@klakhov 

Resolved comments.

> Also what do you think about showing current from value in input(maybe we should disable it in such case) instead of keeping it empty?

I do not think we should disable it because we still should be able to modify them, slider and direction are updated automatically when you change empty input.

**Review by klakhov — APPROVED** [review]

**Comment by klakhov** [comment]
I'm ok with changes. But I see one failed test with rotated images, feels its not related with this pr.

**Comment by bsekachev** [comment]
@klakhov 

OMG, this is already the third unstable test in this PR. 
I tried to fix it making assumption that reason is the same like for previous...

**Pull request merged**